modules/livestore/kafka_client_franz.go

```diff
@@ -0,0 +1,71 @@
+package livestore
+
+import (
+	"context"
+
+	"github.com/go-kit/log"
+	"github.com/grafana/tempo/pkg/ingest"
+	"github.com/twmb/franz-go/pkg/kadm"
+	"github.com/twmb/franz-go/pkg/kgo"
+	"github.com/twmb/franz-go/plugin/kprom"
+)
+
+// FranzKafkaClient wraps a kgo.Client and admin client to implement the KafkaClient interface
+type FranzKafkaClient struct {
```

Comment:
franz kafka was the writer and has nothing to do with kafka the queue technology :). probably should rename this.